Source/Core/Core/IOS/USB/Emulated/Skylanders/SkylanderCrypto.cpp

```diff
@@ -0,0 +1,177 @@
+      0x8DC9, 0x7C26, 0x6C07, 0x5C64, 0x4C45, 0x3CA2, 0x2C83, 0x1CE0, 0x0CC1, 0xEF1F, 0xFF3E,
+      0xCF5D, 0xDF7C, 0xAF9B, 0xBFBA, 0x8FD9, 0x9FF8, 0x6E17, 0x7E36, 0x4E55, 0x5E74, 0x2E93,
+      0x3EB2, 0x0ED1, 0x1EF0};
+
+  u16 crc = init_value;
+
+  for (size_t i = 0; i < data.size(); ++i)
+  {
+    const u16 tmp = (crc >> 8) ^ data[i];
+    crc = (crc << 8) ^ CRC_CCITT_TABLE[tmp];
+  }
+
+  return crc;
+}
+// CRC-64 algorithm that is limited to 48 bits every iteration
+u64 ComputeCRC48(std::span<const u8> data)
+{
+  const u64 polynomial = 0x42f0e1eba9ea3693;
+  const u64 initial_register_value = 2ULL * 2ULL * 3ULL * 1103ULL * 12868356821ULL;
+
+  u64 crc = initial_register_value;
+  for (size_t i = 0; i < data.size(); ++i)
+  {
+    crc ^= (static_cast<u64>(data[i]) << 40);
+    for (size_t j = 0; j < 8; ++j)
+    {
+      if (Common::ExtractBit(crc, 47))
+      {
+        crc = (crc << 1) ^ polynomial;
+      }
+      else
+      {
+        crc <<= 1;
+      }
+    }
+  }
+  return crc & 0x0000FFFFFFFFFFFF;
+}
+u64 CalculateKeyA(u8 sector, std::span<const u8, 0x4> nuid)
+{
+  if (sector == 0)
+  {
+    return 73ULL * 2017ULL * 560381651ULL;
+  }
+
+  std::array<u8, 5> data = {nuid[0], nuid[1], nuid[2], nuid[3], sector};
+
+  u64 big_endian_crc = ComputeCRC48(data);
+  u64 little_endian_crc = Common::swap64(big_endian_crc) >> 16;
+
+  return little_endian_crc;
+}
+void ComputeChecksum(ChecksumType type, const u8* data_start, u8* output)
+{
+  u16 crc = 0xFFFF;
+
+  switch (type)
+  {
+  case ChecksumType::Type0:
+  {
+    std::array<u8, 0x1E> input = {};
+    memcpy(input.data(), data_start, 0x1E);
+
+    crc = ComputeCRC16(crc, input);
+    break;
+  }
+  case ChecksumType::Type3:
+  {
+    std::array<u8, 0x110> input = {};
+    memcpy(input.data(), data_start, 0x20);
+    memcpy(input.data() + 0x20, data_start + 0x30, 0x10);
+
+    crc = ComputeCRC16(crc, input);
+    break;
+  }
+  case ChecksumType::Type2:
+  {
+    std::array<u8, 0x30> input = {};
+    memcpy(input.data(), data_start, 0x20);
+    memcpy(input.data() + 0x20, data_start + 0x30, 0x10);
+
+    crc = ComputeCRC16(crc, input);
+    break;
+  }
+  case ChecksumType::Type1:
+  {
+    std::array<u8, 0x10> input = {};
+    memcpy(input.data(), data_start, 0x10);
+    input[0xE] = 0x05;
+    input[0xF] = 0x00;
+
+    crc = ComputeCRC16(crc, input);
+    break;
+  }
+  case ChecksumType::Type6:
+  {
+    std::array<u8, 0x40> input = {};
+    memcpy(input.data(), data_start, 0x20);
+
+    memcpy(input.data() + 0x20, data_start + 0x30, 0x20);
+
+    input[0x0] = 0x06;
+    input[0x1] = 0x01;
+
+    crc = ComputeCRC16(crc, input);
+    break;
+  }
+  default:
+    break;
+  }
+  memcpy(output, &crc, 2);
+}
+void ComputeToyCode(u64 code, u8* output)
+{
+  if (code == 0)
+  {
+    memcpy(output, "N/A", 3);
+    return;
+  }
+
+  std::array<u8, 10> values = {};
+  std::string toy_code = "";
+
+  static std::string lookupTable = "23456789BCDFGHJKLMNPQRSTVWXYZ";
```

Comment:
Please make this a `constexpr char lookupTable[]` instead of `std::string`, that way you can avoid the runtime initialization.